base/bucket_gocb_test.go

```diff
@@ -1108,3 +1177,151 @@ func CouchbaseTestDeleteDocumentAndXATTR(t *testing.T) {
+		log.Printf("del value: %v, del xattr: %v", delRetrievedVal, delRetrievedXattr)
+
+		log.Printf("Post-delete cas: %v", deleteCas)
+	*/
+}
+
+// TestDeleteDocumentAndUpdateXATTR.  Delete the document body and update the xattr.  Used during SG delete
+func CouchbaseTestRetrieveDocumentAndXattr(t *testing.T) {
+	b := GetBucketOrPanic()
+	bucket, ok := b.(*CouchbaseBucketGoCB)
+	if !ok {
+		log.Printf("Can't cast to bucket")
+		return
+	}
+
+	// 1. Create document with XATTR
+	val := make(map[string]interface{})
+	val["type"] = "docExistsXattrExists"
+
+	xattrName := "_sync"
+	xattrVal := make(map[string]interface{})
+	xattrVal["seq"] = 123
+	xattrVal["rev"] = "1-1234"
+
+	key1 := "DocExistsXattrExists"
+	key2 := "DocExistsNoXattr"
+	key3 := "XattrExistsNoDoc"
+	key4 := "NoDocNoXattr"
+	var err error
+
+	// Create w/ XATTR
+	cas := uint64(0)
+	cas, err = bucket.WriteCasWithXattr(key1, xattrName, 0, cas, val, xattrVal)
+	if err != nil {
+		t.Errorf("Error doing WriteCasWithXattr: %+v", err)
+	}
+
+	// 2. Create document with no XATTR
+	val = make(map[string]interface{})
+	val["type"] = "DocExistsNoXattr"
+	_, err = bucket.Add(key2, 0, val)
+
+	// 3. Xattr, no document
+	val = make(map[string]interface{})
+	val["type"] = "xattrExistsNoDoc"
+
+	xattrVal = make(map[string]interface{})
+	xattrVal["seq"] = 456
+	xattrVal["rev"] = "1-1234"
+
+	// Create w/ XATTR
+	cas = uint64(0)
+	cas, err = bucket.WriteCasWithXattr(key3, xattrName, 0, cas, val, xattrVal)
+	if err != nil {
+		t.Errorf("Error doing WriteCasWithXattr: %+v", err)
+	}
+	// Delete the doc
+	bucket.Delete(key3)
+
+	// 4. No xattr, no document
+
+	// Attempt to retrieve all 4 docs
+	res1, key1err := bucket.Bucket.LookupInEx(key1, gocb.SubdocDocFlagAccessDeleted).
+		GetEx(xattrName, gocb.SubdocFlagXattr). // Get the xattr
+		GetEx("", gocb.SubdocFlagNone).         // Get the document body
+		Execute()
+	log.Printf("key1err: %v", key1err)
+	log.Printf("key1res: %+v", res1)
+
+	time.Sleep(100 * time.Millisecond)
+	res2, key2err := bucket.Bucket.LookupInEx(key2, gocb.SubdocDocFlagAccessDeleted).
+		GetEx(xattrName, gocb.SubdocFlagXattr). // Get the xattr
+		GetEx("", gocb.SubdocFlagNone).         // Get the document body
+		Execute()
+	log.Printf("key2err: %v", key2err)
+	log.Printf("key2res: %+v", res2)
+
+	time.Sleep(100 * time.Millisecond)
+	res3, key3err := bucket.Bucket.LookupInEx(key3, gocb.SubdocDocFlagAccessDeleted).
+		GetEx(xattrName, gocb.SubdocFlagXattr). // Get the xattr
+		GetEx("", gocb.SubdocFlagNone).         // Get the document body
+		Execute()
+	log.Printf("key3err: %v", key3err)
+	log.Printf("key3res: %+v", res3)
+
+	time.Sleep(100 * time.Millisecond)
+	res4, key4err := bucket.Bucket.LookupInEx(key4, gocb.SubdocDocFlagAccessDeleted).
+		GetEx(xattrName, gocb.SubdocFlagXattr). // Get the xattr
+		GetEx("", gocb.SubdocFlagNone).         // Get the document body
+		Execute()
+	log.Printf("key4err: %v", key4err)
+	log.Printf("key4res: %+v", res4)
+
```

Comment:
This entire test needs to be revisited when the code actually works (it's pending https://issues.couchbase.com/browse/MB-24152) - it was added as a way to generate pcap for that ticket, and I left it in there as raw material for the final unit test.  I'll add a comment explaining these details.